qiskit-qir/src/qiskit_qir/visitor.py

```diff
@@ -0,0 +1,151 @@
+        self._module = SimpleModule(
+            name=module.name,
+            num_qubits=module.num_qubits,
+            num_results=module.num_clbits,
+        )
+        self._builder = BasicQisBuilder(self._module.builder)
+
+    def visit_register(self, register):
+        _log.debug(f"Visiting register '{register.name}'")
+        if isinstance(register, QuantumRegister):
+            self._qubit_labels.update({
+                bit: n + len(self._qubit_labels) for n, bit in enumerate(register)
+            })
+        elif isinstance(register, ClassicalRegister):
+            self._clbit_labels.update({
+                bit: n + len(self._clbit_labels) for n, bit in enumerate(register)
+            })
+        else:
+            raise ValueError(f"Register of type {type(register)} not supported.")
+
+    def visit_instruction(self, instruction, qargs, cargs, skip_condition=False):
+        if instruction.name not in SUPPORTED_INSTRUCTIONS:
+            raise ValueError(f"Gate {instruction.name} is not supported. \
+Please transpile using the list of supported gates: {SUPPORTED_INSTRUCTIONS}.")
+
+        qlabels = [self._qubit_labels.get(bit) for bit in qargs]
+        clabels = [self._clbit_labels.get(bit) for bit in cargs]
+        qubits = [self._module.qubits[n] for n in qlabels]
+        results = [self._module.results[n] for n in clabels]
+
+        labels = ", ".join([str(l) for l in qlabels + clabels])
+        if instruction.condition is None or skip_condition:
+            _log.debug(f"Visiting instruction '{instruction.name}' ({labels})")
+
+        if instruction.condition is not None and skip_condition is False:
+            _log.debug(f"Visiting condition for instruction '{instruction.name}' ({labels})")
+            conditions = [self._module.results[self._clbit_labels.get(bit)] for bit in instruction.condition[0]]
+
+            # Convert value into a bitstring of the same length as classical register
+            values = format(instruction.condition[1], f'0{len(results)}b')
+
+            # Add branches recursively for each bit in the bitstring
+            def __visit():
+                self.visit_instruction(instruction, qargs, cargs, skip_condition=True)
+
+            def _branch(conditions_values):
+                try:
+                    result, val = next(conditions_values)
+                    def __branch():
+                        self._builder.if_result(
+                            result=result,
+                            one=_branch(conditions_values) if val == "1" else None,
+                            zero=_branch(conditions_values) if val == "0" else None
+                        )
+                except StopIteration:
+                    return __visit
+                else:
+                    return __branch
+
+            _branch(zip(conditions, values))()
+        elif "measure" == instruction.name or "m" == instruction.name:
```

Comment:
Does the qiskit circuit perform group measurements? Or are we always sure this will be a single qubit and single result?

Reply:
Yes, it does perform group measurements. It looks like pyqir does not, thanks for the heads up